starboard/android/shared/media_codec_bridge_eradicator.cc

```diff
@@ -0,0 +1,108 @@
+// Copyright 2024 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "starboard/android/shared/media_codec_bridge_eradicator.h"
+
+#include <pthread.h>
+
+#include "starboard/android/shared/jni_utils.h"
+#include "starboard/common/log.h"
+#include "starboard/common/once.h"
+
+namespace starboard {
+namespace android {
+namespace shared {
+
+SB_ONCE_INITIALIZE_FUNCTION(MediaCodecBridgeEradicator,
+                            MediaCodecBridgeEradicator::GetInstance);
+
+namespace {
+
+struct EradicateParam {
+  explicit EradicateParam(MediaCodecBridgeEradicator* eradicator,
+                          jobject j_media_codec_bridge,
+                          jobject j_reused_get_output_format_result)
+      : eradicator(eradicator),
+        j_media_codec_bridge(j_media_codec_bridge),
+        j_reused_get_output_format_result(j_reused_get_output_format_result) {}
+  MediaCodecBridgeEradicator* eradicator;
+  jobject j_media_codec_bridge;
+  jobject j_reused_get_output_format_result;
+};
+
+}  // namespace
+
+void MediaCodecBridgeEradicator::CheckEradicatingAndMaybeWait() {
+  ScopedLock scoped_lock(mutex_);
+  while (!j_media_codec_bridge_set_.empty()) {
+    condition_variable_.Wait();
+  }
+}
+
+bool MediaCodecBridgeEradicator::Eradicate(
+    jobject j_media_codec_bridge,
+    jobject j_reused_get_output_format_result) {
+  EradicateParam* param = new EradicateParam(this, j_media_codec_bridge,
+                                             j_reused_get_output_format_result);
+
+  // spawn a thread to run the release method on the MediaCodecBridge java
```

Comment:
Spawn a thread for clean up asynchronously.  We can also remove the comment as it's pretty obvious.